pkg/helm/watcher/cache/cache.go

```diff
@@ -158,6 +155,39 @@ func (c *ProfileCache) ListProfiles(ctx context.Context, helmRepoNamespace, helm
 	return result, nil
 }
 
+// ListAvailableVersionsForProfile returns all stored available versions for a profile.
+func (c *ProfileCache) ListAvailableVersionsForProfile(ctx context.Context, helmRepoNamespace, helmRepoName, profileName string) ([]string, error) {
+	// Because the folders of versions are only stored when there are values for a version
+	// we, instead, look in the profiles.yaml file for available versions.
+	var result []string
+
+	getAllAvailableVersionsOp := func() error {
+		var profiles []*pb.Profile
+		if err := c.getProfilesFromFile(helmRepoNamespace, helmRepoName, &profiles); err != nil {
+			if os.IsNotExist(err) {
+				return nil
+			}
+
+			return fmt.Errorf("failed to read profiles data for helm repo: %w", err)
+		}
+
+		for _, p := range profiles {
+			if p.Name == profileName {
+				result = append(result, p.AvailableVersions...)
+				return nil
+			}
+		}
+
+		return fmt.Errorf("profile with name %s not found in cached profiles", profileName)
+	}
+
+	if err := c.tryWithLock(ctx, getAllAvailableVersionsOp); err != nil {
+		return nil, err
+	}
+
+	return result, nil
```

Comment:
```suggestion
	return result, c.tryWithLock(ctx, getAllAvailableVersionsOp)
```

this would work right? Not saying its cleaner, just thinking

Reply:
technically... but I don't really like that. :D